src/main/kotlin/com/sourcegraph/cody/chat/ChatSession.kt

```diff
@@ -179,6 +201,25 @@ private constructor(private val project: Project, newSessionId: CompletableFutur
     }
   }
 
+  @RequiresEdt
+  private fun addAssistantUsedContextToChat(message: ChatMessage) {
+    synchronized(this) {
+      val findResult = messages.indexOfLast { it.id == message.id }
+      if (findResult >= 0) {
+        messages[findResult] = message
+
+        // The first response does not include the context files. We must check
+        // for the second version of it when it's updated with the proper data.
+        val isFirstUpdate = theMostRecentContextFilesMessageId.getAndSet(message.id) != message.id
```

Comment:
You don't need that class-wide `theMostRecentContextFilesMessageId` variable.
You can just check if last message in `messages` have the same UUID. If it's the same, it's an update.